packages/contracts-bedrock/test/L2/L2ToL2CrossDomainMessenger.t.sol

```diff
@@ -0,0 +1,731 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Testing utilities
+import { Test } from "forge-std/Test.sol";
+import { Vm } from "forge-std/Vm.sol";
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+// Target contract
+import {
+    L2ToL2CrossDomainMessenger,
+    NotEntered,
+    MessageDestinationSameChain,
+    RelayMessageCallerNotCrossL2Inbox,
+    CrossL2InboxOriginNotL2ToL2CrossDomainMessenger,
+    MessageDestinationNotRelayChain,
+    MessageTargetCrossL2Inbox,
+    MessageTargetL2ToL2CrossDomainMessenger,
+    MessageAlreadyRelayed,
+    ReentrantCall
+} from "src/L2/L2ToL2CrossDomainMessenger.sol";
+import { CrossL2Inbox } from "src/L2/CrossL2Inbox.sol";
+
+/// @title L2ToL2CrossDomainMessengerWithModifiableTransientStorage
+/// @dev L2ToL2CrossDomainMessenger contract with methods to modify the transient storage.
+///      This is used to test the transient storage of L2ToL2CrossDomainMessenger.
+contract L2ToL2CrossDomainMessengerWithModifiableTransientStorage is L2ToL2CrossDomainMessenger {
+    /// @dev Sets the cross domain messenger sender in transient storage.
+    /// @param _sender Sender address to set.
+    function setCrossDomainMessageSender(address _sender) external {
+        assembly {
+            tstore(CROSS_DOMAIN_MESSAGE_SENDER_SLOT, _sender)
+        }
+    }
+
+    /// @dev Sets the cross domain messenger source in transient storage.
+    /// @param _source Source chain ID to set.
+    function setCrossDomainMessageSource(uint256 _source) external {
+        assembly {
+            tstore(CROSS_DOMAIN_MESSAGE_SOURCE_SLOT, _source)
+        }
+    }
+
+    /// @dev Sets the entered slot value in transient storage to 1.
+    function setEnteredValueTrue() external {
```

Comment:
I recommend turning this into a function that accepts an arg and then tstores into the correct slot the arg